pkg/module/validation.go

```diff
@@ -3,11 +3,100 @@ package module
 import (
 	"errors"
 	"fmt"
+	"os"
+	"path/filepath"
 	"strings"
+	"time"
 
+	"github.com/kyma-project/cli/internal/kube"
+	"go.uber.org/zap"
 	amv "k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
 )
 
+const defaultCRName = "default.yaml"
+
+//readDefaultCR reads the default CR file's contents. The first returned value is true if the file exists. It's false if the file does not exist or an error occurred.
+func readDefaultCR(modulePath string) (bool, []byte, error) {
+	//TODO: Do we need to override the name or it's always "default.yaml"?
+	crPath := filepath.Join(modulePath, defaultCRName)
+
+	fileInfo, err := os.Stat(crPath)
+	if errors.Is(err, os.ErrNotExist) {
+		return false, nil, nil
+	}
+	if err != nil {
+		return false, nil, fmt.Errorf("Error accessing the default CR file `%q`: %w", crPath, err)
+	}
+
+	if !fileInfo.Mode().IsRegular() {
+		return false, nil, fmt.Errorf("Error reading the default CR file `%q`: Not a regular file", crPath)
+	}
+
+	crData, err := os.ReadFile(crPath)
+	if err != nil {
+		return false, nil, fmt.Errorf("Error reading the default CR file `%q`: %w", crPath, err)
+	}
+
+	return true, crData, nil
+}
+
+func ValidateDefaultCR(modulePath string, log *zap.SugaredLogger) (skipped bool, err error) {
+
+	exists, crData, err := readDefaultCR(modulePath)
+	if err != nil {
+		return false, err
+	}
+	if !exists {
+		//If there's no "default.yaml" file, skip the validation.
+		return true, nil
+	}
+
+	//TODO: How to get this directory path programatically? What's the contract?
```

Comment:
The way to go is to traverse all the files under the `charts` directory of the module path, and find the one that matches GVK of the given default CR.

Reply:
Turned out to be too fragile, YAML parsing was necessary.